app/src/lib/stores/app-store.ts

```diff
@@ -1771,6 +1773,40 @@ export class AppStore extends TypedBaseStore<IAppState> {
     this._updateCurrentPullRequest(repository)
     this.updateMenuItemLabels(repository)
     this._initializeCompare(repository)
+    this.repositories = await this.getRefreshedRepositories(
+      this.repositories,
+      repository
+    )
+  }
+
+  public async refreshAllRepositories() {
+    this.repositories = await this.getRefreshedRepositories(this.repositories)
+    this.emitUpdate()
+  }
+
+  private async getRefreshedRepositories(
+    repositories: ReadonlyArray<Repository>,
+    targetRepo?: Repository
+  ): Promise<ReadonlyArray<Repository>> {
+    if (enableRepoInfoIndicators()) {
+      for (const repo of repositories) {
```

Comment:
:art: you can use `this.repositories` here and avoid passing in the parameter, now that we're no longer mutating the objects we get from the repositories store.